### PR TITLE
Add a wal compression flag in rule and receive

### PR DESCRIPTION
**Pull request by yeya24**
Signed-off-by: yeya24 <yb532204897@gmail.com>

* [x] I added CHANGELOG entry for this change.
* [ ] Change is not relevant to the end user.

## Changes

Add a WAL compression option in rule and receive

### CODE REVIEW

**Comment by yeya24** [comment]
Should it be a flag? Receiver sets this option to `true` by default. Is there any use case that people want to set it to false?

**Comment by bwplotka** [comment]
It is an option in Prometheus, plus false by default. (:

**Comment by yeya24** [comment]
OK. I will add the flag for ruler and receiver.

**Comment by yeya24** [comment]
I have added a flag for rule and receive. Should I add a changelog?

**Comment by squat** [comment]
Definitely worth mentioning in changelog if the default has changed

**Comment by brancz** [comment]
WAL compression is intended to become the default in Prometheus, we just can't do it in Prometheus yet as that would be too breaking of a change. I personally think we should just always have WAL compression enabled, the CPU footprint of receive is so tiny it's the right trade-off in my opinion.

tl;dr I think we should leave things as they are in this regard.

**Comment by bwplotka** [comment]
Should we make it true by default then?

**Comment by brancz** [comment]
I'm not sure this should be configurable at all, but yes if this config option exists then it should definitely be true by default.

File: cmd/thanos/receive.go [hunk 1]
```diff
@@ -73,6 +72,8 @@ func registerReceive(m map[string]setupFunc, app *kingpin.Application) {
 
 	tsdbBlockDuration := modelDuration(cmd.Flag("tsdb.block-duration", "Duration for local TSDB blocks").Default("2h").Hidden())
 
+	walCompression := cmd.Flag("tsdb.wal-compression", "Compress the tsdb WAL.").Default("false").Bool()
```
Comment (bwplotka) [inline cmd/thanos/receive.go]: Let's have true by default then

Comment (yeya24) [reply cmd/thanos/receive.go]: Thanks, done.

Comment (bwplotka) [reply cmd/thanos/receive.go]: We thank you (:
